Support comprehensions inside functions when use strict_undefined flag.

cocolato opened the pull request:
Fixes: https://github.com/sqlalchemy/mako/issues/320

Now the test code works as expected if strict_undefined is set to true:

```python
from mako.template import Template

text = """
<%
    mydict = { 'foo': 1 }

    ## Uncomment the following line to workaround the error
    ##k = None
    def getkeys(x):
        return [ k for k in x.keys() ]
%>

${ ','.join( getkeys(mydict) ) }
"""

tmpl = Template(text=text, strict_undefined=True)
out = tmpl.render()
print(out)
```

output:
```



foo

```

cocolato committed:
support parse comprehensions in function.

zzzeek commented:
wow I dig this, you're into mako now.  great let's run the CI

sqla-tester reviewed (commented):
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision daffd3832b4adab291625e1e6efcdb56f8d0d05e of this pull request into gerrit so we can run tests and reviews and stuff

sqla-tester commented:
New Gerrit review created for change daffd3832b4adab291625e1e6efcdb56f8d0d05e: https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/5147

zzzeek commented:
can you add a changelog file?  thanks

zzzeek commented on test/test_template.py:
```diff
@@ -1717,3 +1717,62 @@ def test_inline_percent(self):
             "bar %% baz",
         ]
+
+    def test_lsitcomp_in_func_strict(self):
```
typo

zzzeek reviewed (changes requested):
this is great.  

it needs:

1. a changelog file, like the other one
2. the fixes: line:
  
```
    Fixes: #320
    Closes: #386
    Pull-request: https://github.com/sqlalchemy/mako/pull/386
    Pull-request-sha: daffd3832b4adab291625e1e6efcdb56f8d0d05e
    
    Change-Id: I0591873a83837f8f35b0963c0536df1e2675012f

```

cocolato committed:
Fixes: #320
Closes: #386
Pull-request: https://github.com/sqlalchemy/mako/pull/386
Pull-request-sha: 7840ad30200226f6f97ebd8a834c9a22e2a78d67

Change-Id: I0591873a83837f8f35b0963c0536df1e2675012f

cocolato commented:
All has been updated.

sqla-tester reviewed (commented):
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision cc6a3e0694fb5615db2c3fec2cd23bc9e8a70066 of this pull request into gerrit so we can run tests and reviews and stuff

sqla-tester commented:
Patchset cc6a3e0694fb5615db2c3fec2cd23bc9e8a70066 added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/5147

sqla-tester commented:
**Michael Bayer** (zzzeek) wrote:

thanks!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/5147

sqla-tester commented:
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/mako/+/5147 has been **merged**. Congratulations! :)
